vane-enchantments/src/main/java/org/oddlama/vane/enchantments/enchantments/Lightning.java

```diff
@@ -33,8 +36,14 @@ public Lightning(Context<Enchantments> context) {
     }
 
     public boolean config_enabled_def() {
-        return false;
+        return true;
```

Comment:
My bad, it seems like I didn't remember this correctly. I'm sorry. I've checked and this only works for regular "modules", i.e. classes that extend `ModuleGroup`. The fix is simple, but let me explain what modules are exactly, as it might be good to understand this for the future (and future readers).

All parts and features of vane are grouped into separate modules. Modules can have config options and language strings associated to them `@Config....`, and in theory each module can be disabled independently of one another. Sometimes we expose this functionality as an additional `enabled` config in that module.

To make it easier to enable related modules, they are put in a hierarchy. This means each module has a context to which it belongs, like a "parent module", but without directly imposing nesting which must be specified explicitly.

Each vane plugin (vane-enchantments) is a module in and of itself. Say the plugin as a config option `my_plugin_option` and we write two features as modules which have the config options `feature_option_a` and `feature_option_b`. When modules are created, they specify the context they belong to in their constructor like `super(context)`. So even if the modules of those features have the plugin as the context, the options will be on the _same level_ in the yaml file, as they belong to the same context:

```yaml
my_plugin_option: "test1"
feature_option_a: "test2"
feature_option_b: "test3"
```

If we want nesting, we create a new namespace subcontext for each feature module by calling `super(context.namespace("feature_a"))`. Then we get:

```yaml
my_plugin_option: "test1"
feature_a:
  feature_option_a: "test2"
feature_b:
  feature_option_b: "test3"
```

A common use case is to have a namespace contexts, which users should be able to enable or disable. That is called a group. The main plugin is always a group, as it has an implicit `enabled` field.

So each module is only active in vane, if its context is enabled. Enabled-ness propagates downwards in the context hierarchy. So whatever the context of a module is exactly, it only counts as enabled, if all parent groups in it's hierarchy are enabled. This makes it so easy to enable or disable related vane features without writing any additional code each time.

In the code we would create a group with `super(context.group("Group name")`, which results in this:

```yaml
my_plugin_option: "test1"
feature_a:
  enabled: true
  feature_option_a: "test2"
feature_b:
  enabled: true
  feature_option_b: "test3"
```

---

These groups basically registers it's own `enabled` config field, and if we want to change the default,
it must be done in that specific group. One can call `context.group("name", default_enabled)` for convenience already. So the merit is that this it isn't part of the class that calls `super(context.....)`, which in our case is the enchantment. And that's why the override didn't work (The enabled config is not directly a part of the enchantment's context).

So we need to set this property when creating the context group for the enchantment. To make this easier, I've pushed a new feature that adds a constructor parameter to specify the default enabled value. This means you should rebase upon the newest commit and remove this function here entirely and instead do this:

```
public Lightning(Context<Enchantments> context) {
    // Don't enable this enchantment by default
    super(context, false);
}
```

I hope that explains things a little. If not please ask me to clarify!